src/main/java/org/glassfish/grizzly/memcached/jmx/GrizzlyMemcachedCache.java

```diff
@@ -0,0 +1,245 @@
+        this.mom = mom;
+        rebuildSubTree();
+    }
+
+    @Override
+    protected void onDeregister(GrizzlyJmxManager mom) {
+        this.mom = null;
+    }
+
+    /**
+     * Returns the Java type of the managed object pool
+     *
+     * @return the Java type of the managed object pool.
+     */
+    @ManagedAttribute(id = "grizzly-memcached-cache-type")
+    @Description("The Java type of the memcached cache implementation being used.")
+    public String getCacheType() {
+        return cache.getClass().getName();
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-transport")
+    @Description("A string representation of the transport object used by this memcached cache.")
+    public String getTransport() {
+        return cache.getTransport().toString();
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-connection-timeout-millis")
+    public long getConnectTimeoutInMillis() {
+        return cache.getConnectTimeoutInMillis();
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-write-timeout-millis")
+    public long getWriteTimeoutInMillis() {
+        return cache.getWriteTimeoutInMillis();
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-response-timeout-millis")
+    public long getResponseTimeoutInMillis() {
+        return cache.getResponseTimeoutInMillis();
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-health-monitor-interval-seconds")
+    public long getHealthMonitorIntervalInSecs() {
+        return cache.getHealthMonitorIntervalInSecs();
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-failover")
+    public boolean isFailover() {
+        return cache.isFailover();
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-prefer-remote-config")
+    public boolean isPreferRemoteConfig() {
+        return cache.isPreferRemoteConfig();
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-zookeeper-supported")
+    public boolean isZooKeeperSupported() {
+        return cache.isZooKeeperSupported();
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-zookeeper-server-list-path")
+    public String getZooKeeperServerListPath() {
+        return cache.getZooKeeperServerListPath();
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-current-server-list-from-zookeeper")
+    public String getCurrentServerListFromZooKeeper() {
+        return cache.getCurrentServerListFromZooKeeper();
+    }
+    @ManagedAttribute(id = "grizzly-memcached-cache-current-server-list")
+    public String getCurrentServerList() {
+        return cache.getCurrentServerList().toString();
+    }
+
+    @ManagedOperation(id = "grizzly-memcached-cache-connection-size", impact = Impact.INFO)
+    @Description("The total number of connections currently idle and active in this pool or a negative value if unsupported.")
+    @ParameterNames({"hostname", "port"})
+    public int getConnectionSize( final String hostname, final int port) {
```

Comment:
If this file is edited - let's remove ` ` before first `final`.

Reply:
fixed.